### ISSUE TITLE
Question about sign of loss

### ISSUE THREAD

**Issue by Saurabh7** [fiveai/on-episodes-fsl#3]
Hi,
Thank you for releasing the code.
I was wondering why is the loss multiplied by -1.
 ```  
      loss = -1 * torch.sum(torch.log(frac[frac >= 1e-10])) / n
```

Since PyTorch minimizes the loss, won't the original frac need to be minimized? ( Numerators are distances between positive matches )
It seems multiplying by -1 would lead to opposite behavior.
Apologies if the answer is obvious.

**Comment by SteinarLaenen** [comment]
Hi,

I do admit it is slightly hidden throughout the code, but `frac` is not the fraction of the true distance values, but rather of 
$e^{-||\mathbf{z}_i - \mathbf{z}_j||^2}$. Minimizing the distance $||\mathbf{z}_i - \mathbf{z}_j||^2$ increases the value of $e^{-||\mathbf{z}_i - \mathbf{z}_j||^2}$.

The exponent turns the Neighborhood Component Analysis (NCA) loss into a likelihood function, which we have to maximize. However, since PyTorch does loss minimization, we instead minimize the negative log likelihood, and hence we multiply by -1.

For some more info you can check page 4 of the [paper](https://arxiv.org/pdf/2012.09831.pdf) where we define the loss function $\mathcal{L}_{\mathrm{NCA}}$, and in general for more references on the NCA loss.

Hope that clarifies it, please don't hesitate to ask more questions!

**Comment by Saurabh7** [comment]
Thank you for the response. I somehow missed that :)
